File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
# Using DP

def solution(n):
    answer = 0
    DP = [0 for i in range(n+2)]
    # 1
    DP[1] = 1
    # 1,1 and 2 두가지 방법
    DP[2] = 2
    for i in range(3, n+1):
        DP[i] = DP[i-2]+DP[i-1]
        
    answer = DP[n]
    return answer%1234567

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        long answer = 0;
	        long [] dp =  new long[n+2];
	        dp[1]=1;
	        dp[2]=2;
	        
	        for(int i=3; i<n+1; i++) {
	        	dp[i] = dp[i-2]%1234567 +dp[i-1]%1234567;
	        	
	        }
	        
	        answer = dp[n];
	        return answer%1234567;
    }
}